core/src/main/java/org/commonjava/indy/core/inject/CassandraNotFoundCache.java

```diff
@@ -0,0 +1,273 @@
+        ResultSet result = session.execute( bound );
+        long count = result.one().get( 0, Long.class );
+        boolean missing = count > 0;
+        logger.trace( "NFC check: {}, missing: {}", resource, missing );
+        return missing;
+    }
+
+    @Override
+    @Measure
+    public void clearMissing( final Location location )
+    {
+        StoreKey key = ( (KeyedLocation) location ).getKey();
+        BoundStatement bound = preparedDeleteByStore.bind( key.toString() );
+        session.execute( bound );
+    }
+
+    @Override
+    @Measure
+    public void clearMissing( final ConcreteResource resource )
+    {
+        StoreKey key = getResourceKey( resource );
+        BoundStatement bound = preparedDelete.bind( key.toString(), resource.getPath() );
+        session.execute( bound );
+    }
+
+    @Override
+    @Measure
+    public void clearAllMissing()
+    {
+        session.execute( "TRUNCATE " + keyspace + ".nfc;" );
+    }
+
+    @Override
+    @Measure
+    public Map<Location, Set<String>> getAllMissing()
+    {
+        return Collections.emptyMap(); // not support
+    }
+
+    @Override
+    @Measure
+    public Set<String> getMissing( final Location location )
+    {
+        logger.debug( "[NFC] getMissing for {}", location );
+        StoreKey key = ( (KeyedLocation) location ).getKey();
+        BoundStatement bound = preparedQueryByStore.bind( key.toString() );
+        ResultSet result = session.execute( bound );
+        int count = 0;
+        Set<String> matches = new HashSet<>();
+        for ( Row row : result )
+        {
+            if ( maxResultSetSize > 0 && count >= maxResultSetSize )
+            {
+                logger.debug( "[NFC] getMissing reach maxResultSetSize: {}", maxResultSetSize );
+                break;
+            }
+            count++;
+            matches.add( row.get( "path", String.class ) );
+        }
+        logger.debug( "[NFC] getMissing complete, count: {}", matches.size() );
+        return matches;
+    }
+
+    @Override
+    @Measure
+    public Map<Location, Set<String>> getAllMissing( int pageIndex, int pageSize )
+    {
+        return getAllMissing();
+    }
+
+    /**
+     * Get missing entries via pagination.
+     * @param location
+     * @param pageIndex starts from 0
+     * @param pageSize how many entries in each page
+     * @return
+     */
+    @Override
+    @Measure
+    public Set<String> getMissing( Location location, int pageIndex, int pageSize )
+    {
+        return getMissing( location );
+    }
+
+    @Override
+    @Measure
+    public long getSize( StoreKey storeKey )
+    {
+        BoundStatement bound = preparedCountByStore.bind( storeKey.toString() );
+        ResultSet result = session.execute( bound );
+        return result.one().get( 0, Long.class );
+    }
+
+    @Override
+    @Measure
+    public long getSize()
+    {
+        return 0; // not support
```

Comment:
Again, is this related to querying across partition keys?

Reply:
For both reasons. A full table scan is very inefficient for Cassandra. And we really don't need such a function for a cache like NFC.